com.unity.netcode.gameobjects/Runtime/Core/NetworkBehaviour.cs

```diff
@@ -129,24 +132,59 @@ internal unsafe void __sendClientRpc(FastBufferWriter writer, uint rpcMethodId,
             };
             int messageSize;
 
+            // We check to see if we need to shortcut for the case where we are the host/server and we can send a clientRPC
+            // to ourself. Sadly we have to figure that out from the list of clientIds :( 
+            ulong serverClientId = ulong.MaxValue;
+
             if (rpcParams.Send.TargetClientIds != null)
             {
                 // Copy into a localArray because SendMessage doesn't take IEnumerable, only IReadOnlyList
                 ulong* localArray = stackalloc ulong[rpcParams.Send.TargetClientIds.Count()];
                 var index = 0;
                 foreach (var clientId in rpcParams.Send.TargetClientIds)
                 {
+                    if (clientId == NetworkManager.ServerClientId)
+                    {
+                        serverClientId = clientId;
+                        continue;
+                    }
+
                     localArray[index++] = clientId;
                 }
+
                 messageSize = NetworkManager.SendMessage(message, networkDelivery, localArray, index, true);
             }
             else if (rpcParams.Send.TargetClientIdsNativeArray != null)
             {
+                foreach (var clientId in rpcParams.Send.TargetClientIdsNativeArray)
+                {
+                    if (clientId == NetworkManager.ServerClientId)
+                    {
+                        serverClientId = clientId;
+                    }
+                }
+
                 messageSize = NetworkManager.SendMessage(message, networkDelivery, rpcParams.Send.TargetClientIdsNativeArray.Value);
             }
             else
             {
-                messageSize = NetworkManager.SendMessage(message, networkDelivery, NetworkManager.ConnectedClientsIds, true);
+                foreach (var clientId in NetworkManager.ConnectedClientsIds)
+                {
+                    if (clientId == NetworkManager.ServerClientId)
+                    {
+                        serverClientId = clientId;
+                    }
+                }
+
+                messageSize = NetworkManager.SendMessage(message, networkDelivery, NetworkManager.ConnectedClientsIds);
+            }
+
+            // If we are a server/host then we just no op and send to ourself
+            if (serverClientId != ulong.MaxValue && (IsHost || IsServer))
```

Comment:
Only question is... we're iterating through all these arrays looking for `clientId == NetworkManager.ServerClientId` to set `serverClientId`... but the value of `serverClientId` will always be `NetworkManager.ServerClientId` or `ulong.MaxValue`. Why not just use a boolean `hasServer` and then as Jesse said, when there's no specific target IDs, just `hasServer = IsHost`?